### PR TITLE
Refactor/feed

### PR DESCRIPTION
**Pull request by MihailMatveichuk**
- delete unnecessary audioRef
- add video files for presentation
- add timeline for video and audio player in messanger
- add copy link function
- add premium modal and fix some showing problems
- add learn more modal

Screenshot_1
Screenshot_2
Screenshot_3
Screenshot_4

### CODE REVIEW

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx [hunk 1]
```diff
@@ -68,11 +78,43 @@ const MatchedProfile: React.FC<UserProfileCardProps> = ({
...
         videoRef.current.play();
         setIsPlaying(true);
+        calculateTimeLineValue();
       }
     }
   };
 
-  const volumeRef = useRef<HTMLInputElement | null>(null);
+  const copyLink = useCallback(
+    (e: React.MouseEvent<HTMLDivElement, MouseEvent>, linkValue: string) => {
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:87]: import { MouseEvent } from 'react';

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css [hunk 2]
```diff
@@ -166,9 +169,56 @@
...
+
+.learnMoreModalHeader {
+  display: flex;
+  flex: 1;
+  justify-content: space-between;
+  margin-bottom: 12px;
+}
+
+.learnMoreModalTitle {
+  font-family: Montserrat;
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css:191]:
```suggestion
  font-family: var(--ff);
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css [hunk 3]
```diff
@@ -166,9 +169,56 @@
...
+  font-size: 22px;
+  font-weight: 600;
+  line-height: 24px;
+  color: #5f4f7f;
+}
+
+.learnMoreModalText {
+  font-family: Montserrat;
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css:199]:
```suggestion
  font-family: var(--ff);
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css [hunk 4]
```diff
@@ -166,9 +169,56 @@
...
+  font-size: 16px;
+  font-weight: 400;
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css:201]: it's default

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css [hunk 5]
```diff
@@ -166,9 +169,56 @@
...
+}
+
+.button {
+  width: 100%;
+  padding: 16px 18px;
+  background: linear-gradient(180deg, #9747ff 0%, #e551e8 100%);
+  border-radius: 5px;
+  border: none;
+  color: #ffffff;
+  font-family: "Montserrat", Arial, sans-serif;
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css:215]:
```suggestion
  font-family: var(--ff);
```

File: src/components/Layouts/MainLayout/MatchingFeed/data.ts
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/data.ts:1]: move data to mock foler

File: src/components/Messanger/Workspace/AudioPlayer/AudioPlayer.tsx [hunk 6]
```diff
@@ -133,6 +138,7 @@ export const AudioPlayer = () => {
           </div>
         )}
       </div>
+      {<TimeLine duration={timeLineTime} />}
```
Comment (akrutsko) [inline src/components/Messanger/Workspace/AudioPlayer/AudioPlayer.tsx:141]:
```suggestion
      <TimeLine duration={timeLineTime} />
```

File: src/helpers/formatTime.ts
Comment (akrutsko) [inline src/helpers/formatTime.ts:1]: there is utils folder in another PR

File: src/components/Messanger/Workspace/AudioPlayer/AudioPlayer.tsx (hunk 6 above)
Comment (MihailMatveichuk) [reply src/components/Messanger/Workspace/AudioPlayer/AudioPlayer.tsx:141]: Согласен, забыл убрать

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css (hunk 2 above)
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.module.css:191]: done

File: src/components/Layouts/MainLayout/MatchingFeed/data.ts
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/data.ts:1]: I don't have a mock folder

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { useState } from 'react';
 
-import logo from '../../assets/ModalAuth/mod_welc-logo.svg';
+import logo from '../../assets/ModalAuth/logo.svg';
```
Comment (akrutsko) [inline src/components/Header/Header.tsx:3]: why it's not imported and used as ReactComponent?

Comment (MihailMatveichuk) [reply src/components/Header/Header.tsx:3]: I did it as it was earlier, i think this is not so important)

**Comment by MihailMatveichuk** [comment]
@akrutsko Thanks a lot for your review 👍

File: src/components/Layouts/MainLayout/MainLayout_Header/MainLayoutHeader.tsx [hunk 8]
```diff
@@ -7,13 +7,13 @@ import calendar from '../../../../assets/feed/calendar.svg';
 import arrow from '../../../../assets/feed/arrow.svg';
 import logo from '../../../../assets/ModalAuth/logo.svg';
 import avatar from '../../../../assets/user/avatarIvan.svg';
-import ModalPremium from './ModalPremium/ModalPremium';
 import ModalNotification from 'components/ModalNotification/ModalNotification';
 import { LanguageSelect } from 'components/Header/LanguageSelect/LanguageSelect';
 
 import styles from './mainLayoutHeader.module.css';
+import { ModalSubscription } from 'components/pages/PaymentPage/components/ModalPayment';
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MainLayout_Header/MainLayoutHeader.tsx:14]: move it to other imports

Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MainLayout_Header/MainLayoutHeader.tsx]: done

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx [hunk 9]
```diff
@@ -152,74 +188,72 @@ const MatchedProfile: React.FC<UserProfileCardProps> = ({
...
+                    handleRangeChange(e);
+                  }}
+                />
+              </div>
+            )}
+            {!volume ? (
+              <SoundOffIcon
+                className={styles.soundIcon}
+                onMouseEnter={() => setIsHovered(true)}
+                onMouseLeave={() => {
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:229]:
```suggestion
                onMouseLeave={() => setTimeout(() => setIsHovered(false), 300)}
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx [hunk 10]
```diff
@@ -152,74 +188,72 @@ const MatchedProfile: React.FC<UserProfileCardProps> = ({
...
-            )}
-            {isPlaying && (
-              <PauseIcon
-                className={`${styles.iconControls}`}
-                onClick={handleVideoClick}
+            ) : (
+              <SoundOnIcon
+                className={styles.soundIcon}
+                onMouseEnter={() => setIsHovered(true)}
+                onMouseLeave={() => {
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:238]:
```suggestion
                onMouseLeave={() => setTimeout(() => setIsHovered(false), 500)}
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx [hunk 11]
```diff
@@ -232,6 +266,20 @@ const MatchedProfile: React.FC<UserProfileCardProps> = ({
             ...
           </div>
         </div>
+        <div className={styles.videoControls}>
+          {hovering && !isPlaying && (
+            <PlayIcon
+              className={`${styles.iconControls}`}
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:272]:
```suggestion
              className={styles.iconControls}
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx [hunk 12]
```diff
@@ -232,6 +266,20 @@ const MatchedProfile: React.FC<UserProfileCardProps> = ({
...
+              onClick={handleVideoClick}
+            />
+          )}
+          {hovering && isPlaying && (
+            <PauseIcon
+              className={`${styles.iconControls}`}
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:278]:
```suggestion
              className={styles.iconControls}
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.tsx [hunk 13]
```diff
@@ -218,64 +125,81 @@ const MatchingFeed = () => {
...
       {showMatchModal && (
         <IsMatchedModal
-          closeMatchedModal={closeMatchedModal}
+          closeMatchedModal={handleCloseMatchModal}
           selectedProfileId={selectedProfileId}
           profilesData={profilesData}
         />
       )}
+      <LikeMatchButton
+        onClick={(event: MouseEvent<HTMLButtonElement>) => {
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.tsx:193]: it should work with infer types
```suggestion
        onClick={(event) => {
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx (hunk 9 above)
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:229]: done

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx (hunk 12 above)
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:278]: done, thanks

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.tsx (hunk 13 above)
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.tsx:193]: agreed

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx [hunk 14]
```diff
@@ -152,74 +188,68 @@ const MatchedProfile: React.FC<UserProfileCardProps> = ({
...
-            {isPlaying && (
-              <PauseIcon
-                className={`${styles.iconControls}`}
-                onClick={handleVideoClick}
+            {!volume ? (
+              <SoundOffIcon
+                className={styles.soundIcon}
+                onMouseEnter={() => setIsHovered(true)}
+                onMouseLeave={() => setTimeout(() => setIsHovered(false), 300)}
+                onClick={() => setVolume(1)}
```
Comment (akrutsko) [inline src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:230]: jfyi,  `() => setVolume(0/1)` will lose volume settings. I've implemented keep volume setting in VideoUploader
```ts
const [isMuted, setIsMuted] = useState(false);

const handleVideoMute = () => {
  if (!videoRef.current) return;

  videoRef.current.muted = !videoRef.current.muted;
  setIsMuted(videoRef.current.muted);
};
  
 <SoundOffIcon onClick={handleVideoMute} />
 <SoundOnIcon onClick={handleVideoMute} />
```

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx (hunk 11 above)
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx]: done

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx (hunk 14 above)
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.tsx:230]: This part of code is not mine

File: src/components/pages/PaymentPage/components/ModalPayment/ModalSubscription.tsx [hunk 15]
```diff
@@ -30,8 +31,13 @@ export const ModalPayment: FC<ModalPaymentProps> = ({
   secondButtonText,
   fcFirstBtn,
   children,
+  onClose,
 }) => {
   const handleCloseModal = () => {
+    if (onClose) {
+      onClose();
+      document.body.style.overflow = 'unset';
```
Comment (Sergei181294) [inline src/components/pages/PaymentPage/components/ModalPayment/ModalSubscription.tsx:39]: эта строка не нужна)

File: src/components/Messanger/Footer/RecordingAudio/recordingAudio.module.css [hunk 16]
```diff
@@ -43,9 +43,9 @@
 
 .timeIsRecording {
   position: absolute;
-  top: 9px;
+  top: 11px;
   left: 37px;
-  color: var(--White, #fff);
+  color: #fff;
   font-family: Montserrat;
```
Comment (daryapatsko) [inline src/components/Messanger/Footer/RecordingAudio/recordingAudio.module.css:49]: font-family: var(--ff);

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.module.css [hunk 17]
```diff
@@ -11,25 +10,39 @@
...
-  width: 324px;
-}
-
-.videoProfile {
-  width: 324px;
-  height: 476px;
+  position: relative;
+  width: 100%;
+  border-bottom-left-radius: 10px;
+  border-bottom-right-radius: 10px;
```
[the remark this answers is not in the document]
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.module.css:17]: в таком случае радиус только слева будет

File: src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.tsx [hunk 18]
```diff
@@ -1,164 +1,67 @@
...
+  //   }
+  // };
+
+  const switchTab = (tab: string) => {
+    setActiveButton(tab);
+
+    if (tab === 'Interested') {
+      setProfilesData(profilesDataInterestedInYou);
+      setIsShowPremiumModal(true);
+    } else if (tab === 'NewPeople') {
```
[the remark this answers is not in the document]
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchingFeed.tsx:58]: done

File: src/components/Layouts/MainLayout/MatchingFeed/components/MatchVideo.tsx [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+          className={`${styles.navCategory} ${
+            activeButton === 'Interested' ? styles.interestedActive : ''
+          }`}
+          onClick={() => switchTab('Interested')}
+        >
+          Interested in you
+        </div>
+        <div
+          className={`${styles.navCategory} ${
+            activeButton === 'NewPeople' ? styles.newPeopleActive : ''
```
[the remark this answers is not in the document]
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/components/MatchVideo.tsx:36]: done

File: src/components/pages/PaymentPage/components/ModalPayment/ModalSubscription.tsx (hunk 15 above)
Comment (MihailMatveichuk) [reply src/components/pages/PaymentPage/components/ModalPayment/ModalSubscription.tsx:39]: done

File: src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.module.css (hunk 17 above)
[the remark this answers is not in the document]
Comment (MihailMatveichuk) [reply src/components/Layouts/MainLayout/MatchingFeed/MatchedProfile/MatchedProfile.module.css:17]: done

**Pull request merged**